minimum_jerk_ros/src/minimum_jerk_ros.cpp

```diff
@@ -0,0 +1,736 @@
+
+            if (goal->enable_collision_check)
+            {
+                if (!check_rotation_collision_(poses, angular_vel))
+                    return;
+            }
+
+            angular_vel.angular.z = double(vel.get_theta());
+            this->publisher_->publish(angular_vel);
+
+            scaled_obstacle_lookahead_distance_ = obstacle_lookahead_distance_ + obstacle_lookahead_distance_ * angular_vel.angular.z;
+
+            auto t_begin_for = get_clock()->now();
+            auto init_timeout = std::chrono::system_clock::now();
+            while (abs(odometry.get_poses().at(i).get_theta()) >= rotation_feedback->angular_distance_traveled - yaw_tolerance)
+            {
+                if (std::chrono::system_clock::now() - init_timeout >= std::chrono::milliseconds(static_cast<int>(1 / this->control_frequency_ * 1000 * 1.15)))
+                    break;
+                rotation_feedback->angular_distance_traveled = compute_angle_();
+
+                if (goal->enable_collision_check)
+                {
+                    if (!check_rotation_collision_(poses, angular_vel))
+                        return;
+                }
+            }
+
+            if (abs(rotation_feedback->angular_distance_traveled - abs(goal->target_yaw)) <= yaw_tolerance)
+            {
+                stop_rotation_(angular_vel);
+                break;
+            }
+
+            action_rotation_server_->publish_feedback(rotation_feedback);
+            if (goal->enable_data_save)
+            {
+                FilFile::fil_one_ligne(fd_for_times, i, double((get_clock()->now() - t_begin_for).nanoseconds()) * 1e-9);
+                FilFile::fil_one_ligne(fd_r_s_times, i, double((get_clock()->now() - t_init_for).nanoseconds()) * 1e-9, odometry.get_timestamps().at(i));
+            }
+            i++;
+        }
+
+        stop_rotation_(angular_vel);
+
+        if (goal->enable_data_save)
+        {
+            FilFile::close_file(fd_for_times);
+            FilFile::close_file(fd_r_s_times);
+        }
+        try
+        {
+            this->current_pose_ = this->get_current_pose_();
+        }
+        catch (tf2::TransformException &e)
+        {
+            action_rotation_server_->terminate_current();
+            return;
+        }
+
+        rotation_feedback->angular_distance_traveled = compute_angle_();
+        auto res = std::make_shared<Rotation::Result>();
+        res->total_elapsed_time.nanosec = int((get_clock()->now() - t_init).nanoseconds());
+        RCLCPP_INFO(this->get_logger(), "distance traveled %f, error %f\n", rotation_feedback->angular_distance_traveled, abs(rotation_feedback->angular_distance_traveled - abs(goal->target_yaw)));
+        (abs(rotation_feedback->angular_distance_traveled - abs(goal->target_yaw)) <= yaw_tolerance) ? action_rotation_server_->succeeded_current(res) : action_rotation_server_->terminate_current();
+    }
+
+    void MinimumJerkRos::translation_callback()
```

Comment:
what happens if both a translation and a rotation goal is sent at the same time ?

I think you need a latch mechanism to reject another goal while one of them is running

(simple bool that is set to true when entering the callback and false when exiting, if it's true when entering the callback then reject the goal as the other one is running)

Reply:
or maybe check the `action_xxxx_server_` handle's documentation to see if there's a public method giving you the running state (might be as simple as `xxx.is_running()`)